### DISCUSSION TITLE
Future of the `COMPRESS_STORAGE` setting in Django 4.2+

### DISCUSSION THREAD

**Discussion by browniebroke** [django-compressor/django-compressor#1199] [Ideas]
Slightly related to #1187, but longer term idea.

Django recently deprecated the `STATICFILES_STORAGE` and `DEFAULT_FILE_STORAGE` setting. In Cookiecutter Django, under some configuration, we do `COMPRESS_STORAGE = STATICFILES_STORAGE`. With the new setting, this would be `COMPRESS_STORAGE = STORAGES["staticfiles"]["BACKEND"]`, which doesn't read as well.

So here is my question: should django-compressor consider adding or migrating the `COMPRESS_STORAGE` setting to something like `COMPRESS_STORAGE_ALIAS`, which would refer to a key from the `STORAGES` dict? 

Obviously, this would only work for Django 4.2+, so it might make sense to wait before the old settings are completely gone from the versions supported by django-compressor, as I imagine reading from 2 possible settings * 2 values (default + user) might make the config code complicated.

**Comment by carltongibson** [comment]
HI @browniebroke — thanks for this. Let's perhaps do this on #1187, which can be the tracking issue. 

I agree adding a `COMPRESS_STORAGE_ALIAS` is the way to go. 

My current half-idea would be to populate that with the `COMPRESS_STORAGE` if not explicit set when the `AppRegistry` is loaded. Then have compressor just use that. 

Following Django's guidance, from the release of Django 5.0 onwards, I'd suggest dropping support for Django <4.2 if there's going to be any complications here.

**Comment by carltongibson** [comment]
@browniebroke I finally got round to drafting what I have in mind here at the DjangoCon Europe sprints. #1248

**Comment by carltongibson** [comment]
I think this should be resolved so I'm going to unpin this.
